src/index.tsx

```diff
@@ -51,7 +51,11 @@ export function ApolloProvider<TCacheShape = any>({
 export function useApolloClient<TCache = object>(): null | ApolloClient<
   TCache
 > {
-  return useContext(ApolloContext);
+  const client = useContext(ApolloContext);
+
+  assertApolloClient(client);
```

Comment:
This change is potentially breaking, but I think it's a good one. I'd even go one step further - inline `assertApolloClient` here and remove `null` as a possible result of  `useApolloClient` (because after that there would always be a correct instance of the client returned or an error is thrown). It would also allow us to replace all occurrences of `client!` with just `client`.

Reply:
I'd just inline the `assertApolloClient` function here and remove it from _utils.ts_. It's used only here and that way we wouldn't have to return `client!`, just `client`.

```suggestion
   if (!client) {
    // https://github.com/apollographql/react-apollo/blob/5cb63b3625ce5e4a3d3e4ba132eaec2a38ef5d90/src/component-utils.tsx#L19-L22
    throw new Error(
      'Could not find "client" in the context or passed in as a prop. ' +
        'Wrap the root component in an <ApolloProvider>, or pass an ' +
        'ApolloClient instance in via props.'
    );
  }
```

Reply:
Agree, I also removed `utils.ts`